core/src/text/textBuffer.cpp

```diff
@@ -91,15 +96,28 @@ bool TextBuffer::addLabel(const TextStyle::Parameters& _params, Label::Transform
     uint32_t stroke = (_params.strokeColor & 0x00ffffff) + (strokeWidth << 24);
 
     for (auto& q : quads) {
+        // the atlas resolution at which the quad uvs have been generated onto
+        glm::vec2 atlasResolution(q.atlasWidth, q.atlasHeight);
+        glm::vec2 uvScaleFactor(1.f, 1.f);
+
+        if (atlasResolution != m_atlasRes) {
+            uvScaleFactor = atlasResolution / m_atlasRes;
+        }
```

Comment:
They won't because the current quad stores the atlas width and height at which it has been generated, if a resize happens while a mesh is built, the quads within a mesh would have stored different atlas sizes; the exact size at which the UVs have been generated against.
A good test case is to start with an atlas size of 2 pixels.